mycroft/skills/skill_loader.py

```diff
@@ -16,20 +16,97 @@
 import gc
 import importlib
 import os
-from os.path import dirname
 import sys
+from os import path, makedirs
 from time import time
 
+import xdg.BaseDirectory
+
+from mycroft.configuration import BASE_FOLDER
 from mycroft.configuration import Configuration
+from mycroft.configuration.ovos import is_using_xdg
 from mycroft.messagebus import Message
+from mycroft.skills.settings import SettingsMetaUploader
 from mycroft.skills.settings import save_settings
 from mycroft.util.log import LOG
 
-from mycroft.skills.settings import SettingsMetaUploader
-
 SKILL_MAIN_MODULE = '__init__.py'
 
 
+def _get_skill_folder_name(conf=None):
+    # TODO deprecate on version 0.0.3 when the warning is no longer needed
+    conf = conf or Configuration.get(remote=False)
+    folder = conf["skills"].get("directory")
+
+    if not folder:
+        # also check under old "msm" section for backwards compat
+        folder = conf["skills"].get("msm", {}).get("directory")
+        if folder:
+            LOG.warning("msm has been deprecated\n"
+                        "please move 'skills.msm.directory' to 'skills.directory' in mycroft.conf\n"
+                        "support will be removed on version 0.0.3")
+        else:
+            folder = "skills"
+    return folder
+
+
+def get_skill_directories(conf=None):
+    # the contents of each skills directory must be individual skill folders
+    # we are still dependent on the mycroft-core structure of skill_id/__init__.py
+
+    conf = conf or Configuration.get(remote=False)
+    folder = _get_skill_folder_name(conf)
+
+    # load all valid XDG paths
+    # NOTE: skills are actually code, but treated as user data!
+    # they should be considered applets rather than full applications
+    skill_locations = list(reversed(
+        [os.path.join(p, folder)
+         for p in xdg.BaseDirectory.load_data_paths(BASE_FOLDER)]
+    ))
+
+    # load the old hardcoded /opt/mycroft/skills folder
+    # assume external tools will install skills here!
+    default = get_default_skills_directory(conf)
+    if default not in skill_locations:
+        skill_locations.append(default)
+
+    # load additional explicitly configured directories
+    conf = conf.get("skills") or {}
+    # extra_directories is a list of directories containing skill subdirectories
+    # NOT a list of individual skill folders
+    skill_locations += conf.get("extra_directories") or []
+    return skill_locations
+
+
+def get_default_skills_directory(conf=None):
```

Comment:
Same as above (looks like these expect a configuration with 'skills')

Reply:
Will add docstrs, but this is all existing functionality not new. The .conf should be heavily commented for reference. This specific function was renamed to be less ambiguos